client/e2e/support/user-list-po.ts

```diff
@@ -20,13 +20,21 @@ export class UserListPage {
     return cy.get('.user-nav-list .user-list-item');
   }
 
-  clickViewProfile(card: Cypress.Chainable<HTMLElement>) {
+  clickViewProfile(card: Cypress.Chainable<JQuery<HTMLElement>>) {
     // return card.element(by.buttonText('VIEW PROFILE')).click();
-    return card.contains('VIEW PROFILE').click();
+    return card.find('button');
```

Comment:
This looks like it finds any button on the card? Isn't that too fragile, why not have it find the specific button as before? Also the function is still called `clickViewProfile` so why are we having it not actually click it anymore?

Reply:
That function in the page objects should probably be renamed. I thought it was more consistent behavior to have the page objects give us elements of the page rather than clicking on a specific thing. Clicking and selecting were both actions that were giving me troubles a bit in Cypress given our use of material and Cypress's reluctance to allow things to be clicked on that were not the right kind of thing. I might have been grasping at straws here. The test this is a part of was giving me a very hard time. I would not be at all surprised if this could be made cleaner. I was having a hard time getting at the text on the button, but it was not the largest of my troubles, so I think I just did this. And, this button might be clickable by Cypress without struggle. Please feel free to refactor and make changes, especially here.

Reply:
Turns out this one wasn't actually being used. I have updated it and updated the spec to use it. See [`bd82636` (#119)](https://github.com/UMM-CSci-3601/3601-iteration-template/pull/119/commits/bd8263693ebd13d0bce34a26bd895f3397c2d8c1)

Reply:
If you provide `.contains()` with both a selector and text to look for it will find the first element *of that selector* that contains the given text. Otherwise it was getting the `span` inside the button which isn't really clickable.

```ts
card.contains('button', 'VIEW PROFILE').click();
```